### ISSUE TITLE
state schema should accept generic dict types as typing.Dict and TypedDict

### ISSUE THREAD

**Issue by ars3nid3** [writer/writer-framework#620]
Hello, really loving the framework you have built here.

I just had a slight issue with the State schema definition as documented [here](https://dev.writer.com/framework/state-schema). It seems that when annotating a variable as a generic dict, an error is thrown by the framework as shown below:
```python
class CustomAppSchema(wf.WriterState):
    foo: dict[str, str]

initial_state = wf.init_state({
    "foo": {
        "1": "Example One",
        "2": "Example Two",
    }
}, schema=CustomAppSchema)
```
Resulting error:
`ValueError: Attribute foo must inherit of State or requires a dict to accept dictionary.`

I found that the error is thrown because `isinstance(foo, dict)` returns True but `issubclass(dict[str, str], dict)` returns False, meaning the framework wants to assign a dict object to foo but did not expect one in `State._set_state_item`.

Is it intended behaviour to reject generic dict typing in the State schema? If so, what is the suggested workaround to still ensure type hints and IDE autocompletion for e.g. a dict[str, str], which is a common data type used as an object in the Repeater components? I currently just revert the generic dict to a normal one in the annotations upon initialization of the schema, but this is of course bad practice I would like to avoid.

Many thanks!

**Comment by FabienArcellier** [comment]
Good catch. We will take a look on it.

**Comment by FabienArcellier** [comment]
it's available in [release `0.8.1`](https://pypi.org/project/writer/)
